compiler/src/scala/quoted/internal/impl/QuoteContextImpl.scala

```diff
@@ -2200,22 +2201,35 @@ class QuoteContextImpl private (ctx: Context) extends QuoteContext, QuoteUnpickl
           case _ => None
     end AmbiguousImplicitsTypeTest
 
+    type Owner = dotc.core.Symbols.Symbol
```

Comment:
I still find this introduces unnecessary conceptual complexity to meta-programmers. The programming model with explicit owners seems to be better and simpler.

Reply:
If we want it explicit we would also need to use it explicitly in the `TreeTraverser`, `TreeMap`, `TreeAccumulator` and `ValDef.let`. We would also need to remove `Symbol.currentOwner`. That would be quite annoying.

Note that the current design already has a contextual owner mechanism but it is hidden behind the `Context` abstraction. No one has understood how to propagate this correctly so far. As this PR shows even I missed the connection between the two concepts in some cases. By making it clear what we actually propagate contextually it was trivially obvious where it was missing.